static/app/views/issueDetails/streamline/foldSection.tsx

```diff
@@ -0,0 +1,84 @@
+import {useCallback} from 'react';
+import styled from '@emotion/styled';
+
+import {IconChevron} from 'sentry/icons';
+import {space} from 'sentry/styles/space';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+const LOCAL_STORAGE_PREFIX = 'issue-details-collapse-';
+export const enum SectionKey {
+  HIGHLIGHTS = 'highlights',
+  STACK_TRACE = 'stack-trace',
+  BREADCRUMBS = 'breadcrumbs',
+  TAGS = 'tags',
+  CONTEXTS = 'contexts',
+}
+
+interface FoldSectionProps {
+  children: React.ReactNode;
+  sectionKey: SectionKey;
+  title: React.ReactNode;
+  initialCollapse?: boolean;
+  preventCollapse?: boolean;
+}
+
+export function FoldSection({
+  children,
+  title,
+  sectionKey,
+  initialCollapse = false,
+  preventCollapse = false,
+  ...props
+}: FoldSectionProps) {
+  const [isCollapsed, setIsCollapsed] = useLocalStorageState(
+    `${LOCAL_STORAGE_PREFIX}${sectionKey}`,
+    initialCollapse
+  );
+
+  const toggleCollapse = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault();
+      setIsCollapsed(!isCollapsed);
+    },
+    [isCollapsed, setIsCollapsed]
+  );
+
+  return (
+    <section {...props}>
+      <Details open={!isCollapsed || preventCollapse}>
+        <Summary
+          preventCollapse={preventCollapse}
+          onClick={preventCollapse ? e => e.preventDefault() : toggleCollapse}
+        >
+          <div>{title}</div>
+          <IconWrapper preventCollapse={preventCollapse}>
```

Comment:
There should be a button somewhere with an appropriate aria label, as-is there aren't any clickable elements

Reply:
`<summary>` is a conventional clickable HTML element that [most often gets implicit roles](https://www.w3.org/TR/html-aria/#docconformance), but there was a note on it:

<img width="815" alt="image" src="https://github.com/user-attachments/assets/4ce5ec0e-8d0a-4591-850b-cb91a7899875">

That said, there's also [an example](https://www.w3.org/TR/html-aria/#side-effects) implying not to change set the role to button, so I guess no action needed here? 

<img width="888" alt="image" src="https://github.com/user-attachments/assets/5f1b1441-8340-4566-b2a3-37ea1e97eb4b">

Though I might be misunderstanding.

Reply:
Ah no you're right, I'm just unfamiliar with the summary element!